pkgs/by-name/jo/joplin-desktop/source/package.nix

```diff
@@ -0,0 +1,217 @@
+      xcbuild
+      buildPackages.cctools
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isLinux [
+      copyDesktopItems
+    ];
+
+  env = {
+    ELECTRON_SKIP_BINARY_DOWNLOAD = 1;
+
+    # Disable scripts for now, so that yarnBerryConfigHook does not try to build anything
+    # before we can patchShebangs additional paths (see buildPhase).
+    # https://github.com/NixOS/nixpkgs/blob/3cd051861c41df675cee20153bfd7befee120a98/pkgs/by-name/ya/yarn-berry/fetcher/yarn-berry-config-hook.sh#L83
+    YARN_ENABLE_SCRIPTS = 0;
+  };
+
+  postPatch = ''
+    # Add vendored lock file due to removed subpackages
+    cp ${./yarn.lock} ./yarn.lock
+    # Fix build error due to removal of app-mobile
+    sed -i '/app-mobile\//d' packages/tools/gulp/tasks/buildScriptIndexes.js
+    # Don't build the default plugins, would require networking. We build them separately.
+    sed -i "/'buildDefaultPlugins',/d" packages/app-desktop/gulpfile.ts
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    unset YARN_ENABLE_SCRIPTS
+
+    for node_modules in packages/*/node_modules; do
+      patchShebangs $node_modules
+    done
+
+    yarn install --inline-builds
+
+    # electronDist needs to be modifiable on Darwin
+    cp -r ${electron.dist} electronDist
+    chmod -R u+w electronDist
+    electronDist="$PWD/electronDist"
+
+    cd packages/app-desktop
+
+    # copy over defaultPlugins
+    ${lib.concatMapStringsSep "\n" (
+      plugin: "install -Dt ./build/defaultPlugins ${plugin}/*.jpl"
+    ) finalAttrs.defaultPlugins}
+
+    # file is expected to be present for Linux build
+    mkdir dist && touch dist/AppImage
+
+    yarn run electron-builder \
+      --dir \
+      -c.electronDist="$electronDist" \
+      -c.electronVersion=${electron.version} \
+      -c.mac.identity=null
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    ${lib.optionalString stdenv.hostPlatform.isLinux ''
+      mkdir -p "$out/share/joplin-desktop"
+      cp -r dist/*unpacked/resources/* "$out/share/joplin-desktop"
+
+      for file in "$src/Assets/LinuxIcons"/*.png; do
+        resolution=$(basename "$file" .png)
+        install -Dm644 "$file" "$out/share/icons/hicolor/$resolution/apps/joplin.png"
+      done
+
+      makeWrapper ${lib.getExe electron} $out/bin/joplin-desktop \
+        --add-flags $out/share/joplin-desktop/app.asar \
+        --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform=wayland --enable-features=WaylandWindowDecorations --enable-wayland-ime}}" \
+        --inherit-argv0
+    ''}
+
+    ${lib.optionalString stdenv.hostPlatform.isDarwin ''
+      mkdir -p $out/Applications
+      cp -r dist/mac*/Joplin.app $out/Applications
+      makeWrapper $out/Applications/Joplin.app/Contents/MacOS/Joplin $out/bin/joplin-desktop
+    ''}
+
+    runHook postInstall
+  '';
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "joplin";
+      desktopName = "Joplin";
+      exec = "joplin-desktop --no-sandbox %U";
```

Comment:
question: why `--no-sandbox`?

Reply:
I took all of that from the current package's desktop file, which comes from the AppImage. It seems to be only relevant for the AppImage, see [here](https://github.com/laurent22/joplin/blob/0fec5779321109a0db6af98db9a1237fc28b4b6d/packages/lib/utils/processStartFlags.ts#L134). In the upstream .deb it's not there. I'll remove it.